confluence-xml/src/test/java/org/xwiki/contrib/confluence/filter/IntegrationTests.java

```diff
@@ -69,5 +69,7 @@ public String answer(InvocationOnMock invocation) throws Throwable
 
         // Unregister all listeners since they are not needed for testing
         componentManager.registerMockComponent(ObservationManager.class);
+        // Replace the MentionMacroConverter with a copy of it that does not generate random anchors.
+        componentManager.registerComponent(MacroConverter.class, "mention", new DeterministicMentionMacroConverter());
```

Comment:
This means that the MentionMacroConverter is not tested at all, which is not great. At the very least, you could make it mostly tested by allowing an extending class to modify just the random behavior instead of completely replacing it.